src/main/java/com/codessquad/qna/controller/UserController.java

```diff
@@ -1,69 +1,53 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-
-import java.util.ArrayList;
-import java.util.List;
+import org.springframework.web.bind.annotation.*;
 
 @Controller
 @RequestMapping("/users")
 public class UserController {
     private Logger logger = LoggerFactory.getLogger(QuestionController.class);
-    private List<User> users = new ArrayList<>();
+
+    @Autowired
+    private UserRepository userRepository;
 
     @PostMapping
-    public String signup(User user) {
-        user.setId(users.size() + 1);
-        users.add(user);
+    public String signUp(User user) {
+        userRepository.save(user);
         logger.info(user.toString());
         return "redirect:/users";
     }
 
     @GetMapping
     public String list(Model model) {
-        model.addAttribute("users", users);
+        model.addAttribute("users", userRepository.findAll());
         return "user/list";
     }
 
-    @GetMapping("/{userId}")
-    public String viewUserProfile(@PathVariable String userId, Model model) {
-        for (User user : users) {
-            if (user.isMatchingUserId(userId)) {
-                model.addAttribute("user", user);
-                return "user/profile";
-            }
-        }
-        return "redirect:/users";
+    @GetMapping("/{id}")
+    public String viewUserProfile(@PathVariable Long id, Model model) {
+        model.addAttribute("user", userRepository.findById(id).get());
+        return "user/profile";
     }
 
     @GetMapping("{id}/form")
-    public String viewUpdateUserForm(@PathVariable int id, Model model) {
-        try {
-            User user = users.get(id - 1);
-            model.addAttribute("user", user);
-            return "user/updateForm";
-        } catch (IndexOutOfBoundsException e) {
-            return "redircet:/users";
-        }
+    public String viewUpdateUserForm(@PathVariable Long id, Model model) {
+        model.addAttribute("user", userRepository.findById(id).get());
+        return "user/updateForm";
     }
 
-    @PostMapping("{id}/update")
-    public String updateUser(@PathVariable int id, User updateUser) {
-        try {
-            User targetUser = users.get(id - 1);
-            targetUser.setUser(updateUser);
-            return "redirect:/users";
-        }
-        catch (IndexOutOfBoundsException e){
-            return "redirect:/{id}/form";
-        }
+    @PutMapping("{id}/update")
+    public String updateUser(@PathVariable Long id, User updateUser) {
+        User targetUser = userRepository.findById(id).get();
+        targetUser.setUser(updateUser);
```

Comment:
```suggestion
        targetUser.update(user);
```
set은 setter 메서드에만 한정에서 사용해주세요.

Reply:
수정 완료했습니다.